src/pages/profile/profile.tsx

```diff
@@ -133,6 +135,25 @@ const Profile: React.FC = () => {
     event.detail.complete(); // Notify the refresher that loading is complete
   }
 
+  // Function to update the step goal in the database
+  const updateStepGoal = async (stepGoal: number) => {
+    const dbRef = doc(FirestoreDB, 'users', auth.currentUser.email as string);
+    await updateDoc(dbRef, { step_goal: stepGoal })
+      .then(() => {
+        alert('Step Goal updated!');
+        history.go(0); //refresh page
```

Comment:
We have to avoid using history.go(0).
Instead, implement useEffect hook that uses onSnapshot

Reply:
You can ignore this part as it has already been addressed in my PR.